tests/e2e/test-snippets/test-snippets.php

```diff
@@ -23,3 +23,20 @@ function ( $modes ) {
 		return $modes;
 	}
 );
+
+/*
+ * Mimic the behavior of Google Listings & Ads or other plugins,
+ * adding some inline events before `wp_enqueue_scripts.`
+ */
+add_action(
+	'wp_head',
```

Comment:
I tested both theme types with the following changes to simulate the implementation before this PR, but both themes can pass, which means the test case cannot assert the purpose.

```diff
diff --git a/includes/class-wc-google-gtag-js.php b/includes/class-wc-google-gtag-js.php
index 212e381..b3ee9e4 100644
--- a/includes/class-wc-google-gtag-js.php
+++ b/includes/class-wc-google-gtag-js.php
@@ -48,7 +48,7 @@ class WC_Google_Gtag_JS extends WC_Abstract_Google_Analytics_JS {
 
                $this->map_hooks();
 
-               $this->register_scripts();
+               // $this->register_scripts();
                // Setup frontend scripts
                add_action( 'wp_enqueue_scripts', array( $this, 'enquque_tracker' ), 5 );
                add_action( 'wp_footer', array( $this, 'inline_script_data' ) );
@@ -89,6 +89,7 @@ class WC_Google_Gtag_JS extends WC_Abstract_Google_Analytics_JS {
         * @return void
         */
        public function enquque_tracker(): void {
+               $this->register_scripts();
                wp_enqueue_script( 'google-tag-manager' );
                // tracker.js needs to be executed ASAP, the remaining bits for main.js could be deffered,
                // but to reduce the traffic, we ship it all together.
```

I guess it's because `wp_enqueue_scripts` is triggered by `wp_head`, so even `wp_head` is run first, callbacks registered to `wp_head` still be run after the ones registered to `wp_enqueue_scripts`.

![1](https://github.com/woocommerce/woocommerce-google-analytics-integration/assets/17420811/e06e81a1-af3d-42a6-86ff-8665d081c885)

I'm not sure which one is better, but it seems that the `wp` action at least ensures that its callbacks are always run earlier than `wp_enqueue_scripts`.